### PR TITLE
change: ensure exit code is an int

### PR DESCRIPTION
**Pull request by ChoiByungWook**
*Description of changes:*
Ensure that the exit_code provided by the final exception handler provided is a valid int as expected by os._exit.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-containers/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-containers/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-containers/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://5jg9tp8agh.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=367a55ca-49a8-4f02-a7cd-68d36c9b6af0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker_containers/_trainer.py [hunk 1]
```diff
@@ -25,6 +25,15 @@
 DEFAULT_FAILURE_CODE = 1
 
 
+def _get_valid_exit_code(exit_code):
```
Comment (laurenyu) [inline src/sagemaker_containers/_trainer.py]: let's rename this so that it's clear this is used only for failures

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://5jg9tp8agh.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=240be0ff-dbdd-4fb2-bf02-8259d8f2e5e5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://5jg9tp8agh.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c5fd4fd2-c3a4-4644-9aa1-1b455a3ab5ee%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://5jg9tp8agh.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=8954b51b-e83c-4517-93a5-19e3bb6e6122%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**